src/components/Reports/ReportsPageRanking/ReportsPageRanking.js

```diff
@@ -11,31 +11,53 @@ const numberFormatOptions = {
   maximumFractionDigits: 2,
 };
 
+const pageSize = 2;
+const initialState = { pages: [], page: 0, loading: true };
+
 const ReportsPageRanking = ({ domainName, dateFrom, dependencies: { datahubClient } }) => {
-  const [state, setState] = useState({ loading: true });
+  const [state, dispatchListEvent] = useReducer((prevState, action) => {
+    switch (action.type) {
+      case 'loadingMore':
+        return { ...prevState, loading: true };
+      case 'moreLoaded':
+        return { pages: [...prevState.pages, ...action.pages], page: prevState.page + 1 };
+      case 'errorOnLoad':
+        return { ...prevState, error: true };
+      default:
```

Comment:
```suggestion
      case 'reset':
      default:
```

Reply:
it's not the same?

Reply:
Yes, but it allows a nicer (explicit and clear) implementation in the cleanup:

![image](https://user-images.githubusercontent.com/1157864/66864385-69f95d80-ef6b-11e9-9c91-c86195be4ef9.png)